src/downshift.js

```diff
@@ -341,11 +341,14 @@ class Downshift extends Component {
     // the input change as soon as possible. This avoids issues with
     // preserving the cursor position.
     // See https://github.com/downshift-js/downshift/issues/217 for more info.
-    if (!isStateToSetFunction && stateToSet.hasOwnProperty('inputValue')) {
-      this.props.onInputValueChange(stateToSet.inputValue, {
-        ...this.getStateAndHelpers(),
-        ...stateToSet,
-      })
+    if (!isStateToSetFunction) {
+      const newStateToSet = this.props.stateReducer(this.state, stateToSet)
```

Comment:
no reason to call this unless `newStateToSet.hasOwnProperty('inputValue')` so I would leave the condition as it was and call this before calling `onInputValueChange`

Reply:
I had it this way originally but ran into some issues since the user can add or remove `inputValue` with the `stateReducer` this ensures that `onInputValueChange` will be called (or not called) if they add / remove it

Reply:
I might add a comment for this because it's a bit of a gotcha